src/mono/mono/metadata/marshal-noilgen.c

```diff
@@ -3,7 +3,8 @@
 #include <mono/metadata/attrdefs.h>
 #include "metadata/marshal-internals.h"
 #include "metadata/marshal.h"
-#include "metadata/marshal-ilgen.h"
+// TODO: PRobably shouldn't directly refer to a header in a component like this?
+#include "component/marshal-ilgen.h"
 #include "utils/mono-compiler.h"
```

Comment:
2 things:

1. all of these should be `mono/metadata/xyz.h` not `metadata/xyz.h`
2. I really don't like that `mono/component/marshal-ilgen.h` contains: (a) declarations of functions that aren't in the component; (b) declarations of _noilgen_ functions.  Please make two separate headers.